include/highfive/H5DataType.hpp

```diff
@@ -142,6 +144,23 @@ class CompoundType : public DataType {
         create(size);
     }
 
+    inline CompoundType(const hid_t hid) {
+        _hid = hid;
```

Comment:
I should probably add a check here if the `hid` is really pointing to a compound type.

Reply:
Thanks. Very nice indeed! If you are feeling like doing it nicer and introduce compile-time checks, I'd suggest to create a wrapper type around hid_t, so `dtype.getId()` returns, e.g. `TypeId`, being the only type accepted by `CompoundType` constructor (since hid can be anything, inc datasets)
The check for a compound type, probably needs to be runtime.

Reply:
That would be quite the breaking change, wouldn't it? Changing the return type of `getId`? It would also break compile time polymorphism for `H5Object`, right? 

That's one point I haven't yet fully understood, how HighFive is implementing this polymorphism.

Reply:
I also don't like breaking changes, we would need to inherit from the currently returned type. Something we can do on a different PR.
Anyway what seems a better idea is to directly accept a `DataType` object instead of a generic hid.